src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,179 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"jenkinsci.org/plugins/durabletask/common"
+)
+
+type Shell int
+
+const (
+	CMD Shell = iota
+	POWERSHELL
+	PWSH
+)
+
+func (shell Shell) String() string {
+	switch shell {
+	case CMD:
+		return "cmd"
+	case POWERSHELL:
+		return "powershell"
+	case PWSH:
+		return "pwsh"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// Launches the script in a new session and waits for its completion.
+func launcher(wg *sync.WaitGroup, exitChan chan bool, shell string,
+	scriptPath string, resultPath string, outputPath string,
+	launchLogger *log.Logger, scriptLogger *log.Logger) {
+
+	defer wg.Done()
+	defer common.SignalFinished(exitChan)
+
+	if _, err := os.Stat(scriptPath); err != nil {
+		if os.IsNotExist(err) {
+			launchLogger.Printf("%s does not exist", scriptPath)
+			return
+		}
+	}
+
+	var scriptCmd *exec.Cmd
+	switch shell {
+	case CMD.String():
+		scriptCmd = exec.Command("cmd.exe", "/C", scriptPath)
+	case POWERSHELL.String():
+		shellCommand := fmt.Sprintf("[Console]::OutputEncoding = [Text.Encoding]::UTF8; .\\%s", scriptPath)
+		scriptCmd = exec.Command("powershell.exe", "-NoProfile", "-NonInteractive", "-ExecutionPolicy", "Bypass", "-Command", shellCommand)
+	case PWSH.String():
+	default:
+		launchLogger.Println("shell type not supported")
+		return
+	}
```

Comment:
If you mean the complexity of the `exec.cmd`, originally there was a wrapper `powershellhelper.ps1` that was guaranteeing that the output would be no-bom utf8. And I think you needed all those additional flags for powershell. Now that I removed the need for `powershellhelper`, I can revisit this to see if it can be simplified. 

Or, did you mean rather that `durable-task` should assemble all the flags and commands so that the wrapper just launches it and forgets? I guess I prefer having that launch complexity in here because should anything change in `go` land, then I would also have to work on changing the implementation in `durable-task` as well as in this repo.

Reply:
Made more generic in e841c56ce60397e7f301fa42531d3f01659a66d0